Firmware/Filament_sensor.h

```diff
@@ -212,7 +185,7 @@ class PAT9125_sensor: public Filament_sensor {
     
     bool jamDetection;
     int16_t oldPos;
-    volatile int16_t stepCount;
+    int16_t stepCount;
```

Comment:
I made it volative since it gets modified outside the main context. The volatile here is just so it doesn't get cached in the main context by the compiler for whatever reason

Reply:
Did you know that _volatile_ is deprecated in the latest C++ standards?

Reply:
I had no idea. I knew you must always use volatile for stuff that might change outside the control of the main context

Reply:
Tbh, that's what volatile used to be abused for back in the days. Now, volatile is deprecated and does not prevent you from running into data races. To be completely sure about the synchronization, it is better to encapsule the variable in question into an ATOMIC_BLOCK (or std::atomic if it was available).
Moreover, volatile forces the compiler to generate read/write instructions for that variable at any access occasion (that's where the "out of context" approach derived from and magically worked on single processor platforms).